aiosmtpd/handlers.py

```diff
@@ -148,13 +148,34 @@ def process_message(self, peer, mailfrom, rcpttos, data, **kws):
         message['X-Peer'] = str(peer)
         message['X-MailFrom'] = mailfrom
         message['X-RcptTos'] = COMMASPACE.join(rcpttos)
+
+        return message
+
+    def process_message(self, peer, mailfrom, rcpttos, data, **kws):
+        message = self.prepare_message(peer, mailfrom, rcpttos, data, **kws)
```

Comment:
I moved code from `process_message()` into separated `prepare_message()` for reuse for `AsyncMessage` and avoid copy-past.